pymongo/uri_parser.py

```diff
@@ -539,7 +540,9 @@ def parse_uri(
         fqdn, port = nodes[0]
         if port is not None:
             raise InvalidURI(f"{SRV_SCHEME} URIs must not include a port number")
-
+        for host in [node[0] for node in nodes]:
+            if _detect_external_db(host):
```

Comment:
Can we remove this and have _detect_external_db only be called once in MongoClient?

Reply:
Unless we want to refactor our SRV/URI logic to always pull out hosts as a separate step in MongoClient.

Reply:
The ticket specifies that we do so: https://jira.mongodb.org/browse/PYTHON-3835.

Reply:
I'd prefer to ignore that guidance. The practical issue is that for MongoClient() we want the warning.warn to have the correct stacklevel so that the warning is reported for the line that creates the client.

Reply:
I don't quite understand how the stacklevel for `warning.warn` relates to the SRV lookup timing, can you elaborate?

Reply:
The stacklevel for the uri parser will need to be different than the stack level for the client. So the warning will show up differently if you pass a mongodb:// uri or a mongodb+srv:// uri which seems odd. Ideally the warning is always in once place and the stack level make it so that it gets reported on the line that created the MongoClient().

Reply:
For example play around with a program like this:
```
client1 = MongoClient("mongodb://...")
client2 = MongoClient("mongodb+srv://...")
```

And see what warnings are reported.

Reply:
This does make testing more difficult, as now we need an actual live host for the resolver to hit. That doesn't seem like a solvable issue for a unit or even an integration test.